### ISSUE TITLE
Components `folders` config isn't additive contrary to what the docs say

### ISSUE THREAD

**Issue by hteumeuleu** [maizzle/framework#1057]
- Maizzle Version: 4.5.0
- Node.js Version: 18.16.0

The documentation for the components `folders` configuration (https://maizzle.com/docs/configuration/components) states that “_The paths you defined will be added to the default folders._” This is not what happens in my experience.

What I did:

1. `npx degit maizzle/maizzle my-project` and `npm i`
2. In `src`, add an `includes` folder with a file named `header.html`.
3. In the `promotional.html` template, add an `<x-header>` tag.
4. In `config.js`, add the following config under `build`:

```js
    components: {
      folders: ['src/includes'],
    },
```
5. `npm run build`

Result: 

```
✖ Failed to compile template: promotional.html
⚠ [components] <x-v-fill> could not find v-fill.html in the defined root paths (src/includes, src/layouts, src/templates, src/includes)
```

The components in `src/components/` can no longer be found. I need to manually add all the default folders for it to work properly.

**Comment by cossssmin** [comment]
Thanks for catching this, there's an issue with `lodash.merge` and how it merges the default components config with the user-provided one. Will fix it in today's release.

**Comment by cossssmin** [comment]
Coming back to this as the custom implementation of `merge` has an issue that results in duplication of array items. 

Specifically, something like this will be duplicated at least once, and it goes up depending on the number of front matter variables (I know it sounds weird, it's because of a stupid for loop):

```js
module.exports = {
  locals: {
    columns: [
      {
          title: "one",
      },
      {
          title: "two",
      },
      {
          title: "three",
      },
      {
          title: "four",
      },
    ]
  },
}
```

This can be a problem if you're looping through data, as it will render `title` at least 8 times instead of 4.

I've tested using `lodash.merge` again with your example and it works fine, it may have been fixed in a dependency update in `posthtml-component`. Will test some more and will probably revert in the next release
